explorer/interpreter/type_checker.cpp

```diff
@@ -987,8 +1003,13 @@ auto TypeChecker::ArgumentDeduction::Deduce(Nonnull<const Value*> param,
       if (arg->kind() != Value::Kind::PointerType) {
         return handle_non_deduced_type();
       }
-      return Deduce(&cast<PointerType>(*param).type(),
-                    &cast<PointerType>(*arg).type(),
+      const auto& param_pointed = cast<PointerType>(param)->type();
+      const auto& arg_pointed = cast<PointerType>(arg)->type();
+      if (const auto* arg_class = dyn_cast<NominalClassType>(&arg_pointed);
+          arg_class && arg_class->InheritsClass(&param_pointed)) {
+        return Success();
+      }
```

Comment:
This should only be permitted when `allow_implicit_conversion` is `true`.

Reply:
Good point, done.